### ISSUE TITLE
Can not set bookmarks using the mouse

### ISSUE THREAD

**Issue by rovf** [geany/geany#1699]
I can set a bookmark using the keyboard shortcut control-m, but I can not set it with the mouse. When I move the mouse to the left margin (i.e. where the line numbers are displayed) and do a context-click (left-click), I get the normal context menu, and this does not provide a "set bookmark" entry.

Using Geany 1.30.1 on Windows 7.

**Comment by kugel-** [comment]
On my system right click opens the context menu. Left click adds a bookmark (adds the small + sign)

**Comment by rovf** [comment]
This gave a hint to a problem. First, I confused in my description "left" and "right" click, because I have set my mouse to left handed. More importantly, I found that with what is commonly known as "left click", it indeed DOES add a bookmark, but the problem is that the "left margin" is not, as I thought, the area with the line numbers, but a very tiny (small) area between the line number and the edit pane. Since it also is coloured the same as the line number area, it is virtually impossible to see. I just found it by accident.

**Comment by elextr** [comment]
Yes, its the markers margin, for showing the markers, so its sized to fit the markers and nothing else.

**Comment by rovf** [comment]
Is there a way to have it coloured differently, so that I can at least see it? If not, this would be a feature request.  It would also be convenient to be able to style the line number area (foreground/background colour).

**Comment by elextr** [comment]
AFAIK Geany has no way to set the background colour of any margin except the folding margin, but I believe its available from Scintilla, so a pull request adding it to the colour schemes would probably be accepted.

**Comment by andy5995** [comment]
> I can set a bookmark using the keyboard shortcut control-m, but I can not set it with the mouse.

This is an issue for me when the debugger plugin is active. Left-clicking will cause a red diamond to be displayed in the area where the green plus sign should show, which contradicts what the docs say.

On Geany 1.29.. Debian Stretch and Buster. (I'm working on building the debugger plugin from git and testing it on the latest git version of Geany to see if the behavior is the same)

**Comment by elextr** [comment]
@andy5995 not sure if this is the cause, but there is nothing stopping multiple plugins registering different icons for the same marker, so if you have multiple plugins that use the marker margin running at the same time they might tread on each others icons.

**Comment by elextr** [comment]
Basically nothing manages resources shared between plugins, and most are written as if they own the resource, markers, signals, and a bunch of other things.  There have been discussions about how to handle it, but no resolution to date.

**Comment by elextr** [comment]
And a final thought, sharing clicks on the markers margin is hard because there is nothing to indicate which functionality the user wanted with that click, bookmark, or debugger.
